runtime/src/accounts_db.rs

```diff
@@ -1874,6 +1874,27 @@ pub(crate) struct ShrinkAncientStats {
     pub(crate) ancient_scanned: AtomicU64,
 }
 
+#[derive(Debug, Default)]
+pub(crate) struct ShrinkStatsSub {
+    store_accounts_timing: StoreAccountsTiming,
+    rewrite_elapsed_us: u64,
+    create_and_insert_store_elapsed_us: u64,
+}
+
+impl ShrinkStatsSub {
+    #[allow(dead_code)]
+    pub(crate) fn accumulate(&mut self, other: &Self) {
+        self.store_accounts_timing
+            .accumulate(&other.store_accounts_timing);
+        self.rewrite_elapsed_us = self
+            .rewrite_elapsed_us
+            .saturating_add(other.rewrite_elapsed_us);
+        self.create_and_insert_store_elapsed_us = self
+            .create_and_insert_store_elapsed_us
+            .saturating_add(other.create_and_insert_store_elapsed_us);
```

Comment:
If you include this macro you can do this, which is a bit more succinct:
```rust
        saturating_add_assign!(self.rewrite_elapsed_us, other.rewrite_elapsed_us);
        saturating_add_assign!(self.create_and_insert_store_elapsed_us, other.create_and_insert_store_elapsed_us);
```

Reply:
done